_posts/2024-08-02-gsoc2024_talagayev.md

```diff
@@ -0,0 +1,193 @@
+---
+layout: post
+title: GSoC 2024 - 2D visualization for small molecules
+---
+
+**Contributor**: Valerij Talagayev ([@talagayev](https://github.com/talagayev))
+
+**Mentors**: Cédric Bouysset ([@cbouy](https://github.com/cbouy)), Richard Gowers ([@richardjgowers](https://github.com/richardjgowers)) and Yuxuan Zhuang ([@yuxuanzhuang](https://github.com/yuxuanzhuang))
+
+**Organization**: [MDAnalysis](https://www.mdanalysis.org/)
+
+**Release**: [GSoC 2024: 2D visualization for small molecules Release](https://github.com/talagayev/MDonatello/releases/tag/0.0.1)
+
+During the Google Summer of Code 2024 program I was working on the [2D Visualization of small molecules](https://summerofcode.withgoogle.com/programs/2024/projects/sfy3kuqc), which is an important project, that would allow people to easily visualize the molecule that they have in their file
+through the selection of the molecule as an `AtomGroup` in `MDAnalysis`.
+
+## Goals of the project
+
+The goal of the project was to create a code, that would allow the user to select an `AtomGroup` with the help of `MDAnalysis`, which would be used with an [RDKit](https://www.rdkit.org/) Converter to convert the molecule into an rdkit mol object, which would be used for the visualization.
+
+My Proposal consisted of the following steps:
+
+- Select the AtomGroup via an input from the user via `u = mda.Universe("input.pdb")` to create an MDAnalysis Universe and `ag = u.select_atoms("resname UNK")` to select the `AtomGroup` 
+with the molecule that needs to be displayed
+- Convert the selected AtomGroup into an rdkit mol object via `ag.convert_to("RDKit")`
+- Apply `Chem.RemoveHs` and `AllChem.Compute2DCoords` to obtain a 2D visualization of the molecule
+- Display the molecule via `IpyWidgets` with different checkboxes, that by selection would highlight specific parts of the molecule
+
+After discussion with the mentors and suggestions regarding how the code should be structured, it was decided that I would proceed with this suggestion and would apply the following steps with later on adding the additional
+features in the `IpyWidgets` visualization, which would give the user important information in regards to the characteristics of the molecule that is being visualized.
+
+## What I did
+
+I would divide this section into separate parts, each highlighting one of the important aspects of the work.
+
+### Main Visualization Code
+
+The main part of the code consisted of setting up the code for the visualization, which would allow the display of the molecule in a jupyter notebook. This is performed in the 
```

Comment:
That section is a bit heavy to read, here's a suggestion:
```suggestion
The main part of the code consisted of setting up the widget for the visualization, which would allow the display of the molecule in a jupyter notebook. This is performed in the 
```